pkg/deploy/cmd/delete.go

```diff
@@ -31,34 +33,36 @@ type DeploymentConfigReaper struct {
 	pollInterval, timeout time.Duration
 }
 
-type updateConfigFunc func(d *deployapi.DeploymentConfig)
-
-// updateConfigWithRetries will try to update a deployment config and ignore any update conflicts.
-func updateConfigWithRetries(dn appsinternal.DeploymentConfigsGetter, namespace, name string, applyUpdate updateConfigFunc) (*deployapi.DeploymentConfig, error) {
-	var config *deployapi.DeploymentConfig
-
-	resultErr := retry.RetryOnConflict(retry.DefaultBackoff, func() error {
-		var err error
-		config, err = dn.DeploymentConfigs(namespace).Get(name, metav1.GetOptions{})
-		if err != nil {
-			return err
-		}
-		// Apply the update, then attempt to push it to the apiserver.
-		applyUpdate(config)
-		config, err = dn.DeploymentConfigs(namespace).Update(config)
-		return err
-	})
-	return config, resultErr
-}
-
 // pause marks the deployment configuration as paused to avoid triggering new
 // deployments.
 func (reaper *DeploymentConfigReaper) pause(namespace, name string) (*deployapi.DeploymentConfig, error) {
-	return updateConfigWithRetries(reaper.appsClient.Apps(), namespace, name, func(d *deployapi.DeploymentConfig) {
-		d.Spec.RevisionHistoryLimit = kutil.Int32Ptr(0)
-		d.Spec.Replicas = 0
-		d.Spec.Paused = true
-	})
+	config, err := reaper.appsClient.Apps().DeploymentConfigs(namespace).Get(name, metav1.GetOptions{})
```

Comment:
@mfojtik you don't need to call API to create a patch. This just adds latency.

Reply:
@tnozicka i need fresh version of config

Reply:
I don't think you need the fresh or any version of config to send a patch. 
What data from the fresh config are you using to generate string `{"spec":{"paused":true,"replicas":0,"revisionHistoryLimit":0}}`?

The only case I can think of would be when you would need to have UID as part of the patch and you don't. The reason you are using it seems to be just for convenience and that doesn't justify adding this latency. For users having the cluster on the other side of the pond this is noticeable (can be few seconds).

Reply:
you can do the same with hardcoding the string or CreateTwoWayMergePatch on artificial object

Reply:
i see what you mean, I can hardcode the string, which means that when we change the field name it will fail :-) but I guess for this case it will be ok to do that.

Reply:
If you would want to avoid, create empty object (call it old) and assign those 3 properties to different values and you can use the same approach with CreateTwoWayMergePatch